Assets/__Scripts/MapEditor/UI/Bookmarks/BookmarkContainer.cs

```diff
@@ -44,7 +46,8 @@ public void OnPointerClick(PointerEventData eventData)
                 PersistentUI.Instance.ShowDialogBox("Mapper", "bookmark.delete", HandleDeleteBookmark, PersistentUI.DialogBoxPresetType.YesNo);
                 break;
             case PointerEventData.InputButton.Right:
-                PersistentUI.Instance.ShowInputBox("Mapper", "bookmark.update.dialog", HandleNewBookmarkName, null, data._name);
+                if (Input.GetKey(KeyCode.LeftShift)) PersistentUI.Instance.ShowColorInputBox("Mapper", "bookmark.update.color", HandleNewBookmarkColor, GetComponent<Image>().color);
```

Comment:
We don't use Input.GetKey here 🤨
If you're not going to make this individually rebindable at least use `CMInput.IUtilsActions` that supplies binds for Ctrl, Shift and Alt

Reply:
How would I use `Cminput.IUtilsActions`? I can't seem to figure out how it works and I don't know anything about input systems. I tried looking at other places it's used but it doesn't make sense to me.

Reply:
You just implement the interface

Reply:
Unfortunately, due to the fact that bookmarks are created/removed dynamically, it wouldn't immediately work with the [`CMInputCallbackInstaller`](https://github.com/Caeden117/ChroMapper/blob/dev/Assets/__Scripts/Input/ChroMapper/CMInputCallbackInstaller.cs). This class is what allows the Input System to be as easy as "just implement the interface", but currently only works on MonoBehaviors that exist from the beginning.

To get the input interface to bind here, we'd either need to move this bookmark functionality out to the `BookmarkManager` which *would* be picked up by the callback installer, OR we would have to publicly expose the `FindAndInstallCallbacksRecursive` method (or create a similar method for one singular monobehaviour/transform) and dynamically install callbacks for each `BookmarkContainer`.

Personally I'd go for the former since it's less work to do, and have the `OnPointerClick` code simply forward to the `BookmarkManager`.

Reply:
Is this how the interface would be implemented? 
```cs
public class BookmarkManager : MonoBehaviour, CMInput.IBookmarksActions, CMInput.IUtilsActions
...
    public void OnControlModifier(InputAction.CallbackContext context)
        {
        }

    public void OnAltModifier(InputAction.CallbackContext context)
        {
        }

    public void OnShiftModifier(InputAction.CallbackContext context)
        {
            //Do things
        }

    public void OnMouseMovement(InputAction.CallbackContext context)
        {
        }
```
I tried doing that with a `Debug.Log` to see if anything was happening when I pressed the shift key and nothing was logged, so I think I did it wrong but this was the best I could figure out from looking at other places this stuff was used

Reply:
That *should* work, make sure you didnt have `Log`s suppressed in the Editor.

Reply:
I closed Unity and came back to it later and it works now. Not a clue as to why. Either way thanks for the help and info and I'll see if I can do something useful with this